static/app/views/alerts/rules/metric/presetSidebar.tsx

```diff
@@ -0,0 +1,138 @@
+import {useState} from 'react';
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization, Project} from 'sentry/types';
+import useApi from 'sentry/utils/useApi';
+
+import {Preset, PRESET_AGGREGATES, PresetContext} from './presets';
+
+type Props = {
+  organization: Organization;
+  project: Project;
+  className?: string;
+  onSelect?(preset: Preset, ctx: PresetContext): void;
+  selectedPresetId?: string;
+};
+export default function PresetSidebar(props: Props) {
+  return (
+    <div className={props.className}>
+      <Header>{t('Suggested Alerts')}</Header>
+      {PRESET_AGGREGATES.map((preset, i) => (
+        <PresetSidebarItem
+          key={preset.id}
+          preset={preset}
+          index={i}
+          organization={props.organization}
+          project={props.project}
+          selected={props.selectedPresetId === preset.id}
+          onClick={ctx => props.onSelect && props.onSelect(preset, ctx)}
+        />
+      ))}
+    </div>
+  );
+}
+
+function PresetSidebarItem(props: {
+  index: number;
+  organization: Organization;
+  preset: Preset;
+  project: Project;
+  onClick?: (ctx: PresetContext) => void;
+  selected?: boolean;
+}) {
+  const theme = useTheme();
+  const api = useApi();
+  const [loading, setLoading] = useState(false);
+  const iconColor = theme.charts.getColorPalette(PRESET_AGGREGATES.length)[props.index];
```

Comment:
@Neo-Zhixing Why not just pass the `index` to the styled component and let that decide the color instead of having to deal with the theme inside the component?